akka-stream/src/main/boilerplate/akka/stream/FanInShape.scala.template

```diff
@@ -54,28 +56,42 @@ object UniformFanInShape {
 }
 
 class UniformFanInShape[-T, +O](val n: Int, _init: FanInShape.Init[O]) extends FanInShape[O](_init) {
+
+  //ports get added to `FanInShape.inlets` as a side-effect of calling `newInlet`
+  for (i <- 0 until n) newInlet[T](s"in$i")
+
   def this(n: Int) = this(n, FanInShape.Name[O]("UniformFanIn"))
   def this(n: Int, name: String) = this(n, FanInShape.Name[O](name))
   def this(outlet: Outlet[O], inlets: Array[Inlet[T]]) = this(inlets.length, FanInShape.Ports(outlet, inlets.toList))
   override protected def construct(init: FanInShape.Init[O @uncheckedVariance]): FanInShape[O] = new UniformFanInShape(n, init)
   override def deepCopy(): UniformFanInShape[T, O] = super.deepCopy().asInstanceOf[UniformFanInShape[T, O]]
 
-  val inSeq: immutable.IndexedSeq[Inlet[T @uncheckedVariance]] = Vector.tabulate(n)(i => newInlet[T](s"in$i"))
-  def in(n: Int): Inlet[T @uncheckedVariance] = inSeq(n)
+  final override def inlets: immutable.Seq[Inlet[T @uncheckedVariance]] = super.inlets.asInstanceOf[immutable.Seq[Inlet[T]]]
+
+  @deprecated("Use `inlets` or `in(id)` instead.", "2.5.4")
+  lazy val inSeq: immutable.IndexedSeq[Inlet[T @uncheckedVariance]] = inlets.toIndexedSeq
+  def in(n: Int): Inlet[T @uncheckedVariance] = inlets(n)
```

Comment:
is it know that `inlets` is array or Vector, so we don't do this on List?

Reply:
Yes, it's always a `Vector`, although this relies on the private implementation of its superclass. With binary compatibility being the millstone that it is, I think adding a comment to that effect is the best we can do to prevent someone from inadvertently changing it.